embeds/ge2017-nugget/_source/main.js

```diff
@@ -1,88 +1,122 @@
-console.log('hellloo:');
-
-function findTargetContainer(container) {
-    var prev = container.previousElementSibling;
-    var classes = prev.classList;
-    if (classes.contains('fc-container--thrasher') || classes.contains('fc-container--video') || classes.contains('fc-container__mpu--mobile') || prev.getAttribute('id')=='manchester-attack' || prev.offsetHeight < 260) {
-        var target = findTargetContainer(prev);
-    } else {
-        var target = prev;
-    }
-    return target;
-}
+setTimeout(function() {
 
-function closest(el, selector) {
-    var matchesFn;
+    function findTargetContainer(container) {
+        var prev = container.previousElementSibling;
+        var classes = prev.classList;
+        if (classes.contains('fc-container--thrasher') || classes.contains('fc-container--video') || classes.contains('fc-container__mpu--mobile') || prev.getAttribute('id')=='manchester-attack' || prev.offsetHeight < 260) {
+            var target = findTargetContainer(prev);
+        } else {
+            var target = prev;
+        }
+        return target;
+    }
 
-    // find vendor prefix
-    ['matches','webkitMatchesSelector','mozMatchesSelector','msMatchesSelector','oMatchesSelector'].some(function(fn) {
-        if (typeof document.body[fn] == 'function') {
-            matchesFn = fn;
-            return true;
+    function closest(el, selector) {
+        var matchesFn;
+
+        // find vendor prefix
+        ['matches','webkitMatchesSelector','mozMatchesSelector','msMatchesSelector','oMatchesSelector'].some(function(fn) {
+            if (typeof document.body[fn] == 'function') {
+                matchesFn = fn;
+                return true;
+            }
+            return false;
+        })
+
+        var parent;
+
+        // traverse parents
+        while (el) {
+            parent = el.parentElement;
+            if (parent && parent[matchesFn](selector)) {
+                return parent;
+            }
+            el = parent;
         }
-        return false;
-    })
 
-    var parent;
+        return null;
+    }
 
-    // traverse parents
-    while (el) {
-        parent = el.parentElement;
-        if (parent && parent[matchesFn](selector)) {
-            return parent;
+    var SUPPORTER_COOKIE = "gu_paying_member";
+
+    var CONTRIBUTION_COOKIE = "gu.contributions.contrib-timestamp";
+
+    // Taken from https://www.w3schools.com/js/js_cookies.asp
+    function getCookie(cname) {
+        var name = cname + "=";
+        var decodedCookie = decodeURIComponent(document.cookie);
```

Comment:
Does this need to be wrapped in a `try .. catch`?, it will throw an error if the cookie is malformed ? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent